src/parse.rs

```diff
@@ -105,7 +113,20 @@ pub fn parse(
         return Err(msg.into());
     }
 
-    Ok((Document::new(nodes), links))
+    Ok((Document::new(nodes, newline), links))
+}
+
+fn detect_newline(text: &str) -> &'static str {
+    NEWLINE_REGEX.captures_iter(text).next().map_or_else(
```

Comment:
Wouldn't it be simpler to just check `text.contains("\r\n")` and assume Unix line endings otherwise?

Reply:
(The mixed cases are wrong either way, aren't they?)

Reply:
Yes, mixed is evil! 😈 

I wanted to simply use the first newline detected, as `text.contains("\r\n")` would use win endings as soon as any "wrong" line ending is found in a unix-style file.

I just completed to re-do everything on a separate, clearer branch, so I will move everything to another MR and close this one.

Reply:
Changed as suggested, in #122

Reply:
But see https://github.com/mlange-42/yarner/pull/122#discussion_r578768880